src/conversions/std/num.rs

```diff
@@ -106,17 +106,37 @@ macro_rules! int_fits_c_long {
 
         impl<'source> FromPyObject<'source> for $rust_type {
             fn extract(obj: &'source PyAny) -> PyResult<Self> {
-                let ptr = obj.as_ptr();
-                let val = unsafe {
-                    let num = ffi::PyNumber_Index(ptr);
-                    if num.is_null() {
-                        Err(PyErr::fetch(obj.py()))
+                let val: c_long;
+                // with python 3.8+ PyLong_AsLong takes care of calling PyNumber_Index itself
+                // hence no need for the slow `PyNumber_Index` path
+                // see https://github.com/PyO3/pyo3/pull/3742 for detials.
+                #[cfg(Py_3_8)]
+                {
+                    val = err_if_invalid_value(obj.py(), -1, unsafe {
+                        ffi::PyLong_AsLong(obj.as_ptr())
+                    })?;
+                }
+                #[cfg(not(Py_3_8))]
```

Comment:
Since both paths compile on 3.8, I think a `if cfg!(Py_3_8) { .. } else { .. }` would be more readable.

Reply:
changed.